lms/djangoapps/teams/csv.py

```diff
@@ -19,3 +27,199 @@ def load_team_membership_csv(course, response):
+        self.teamset_ids = []
+        self.number_of_record_added = 0
+        # stores the course module that will be used to get course metadata
+        self.course_module = ''
+        # stores the course for which we are populating teams
+        self.course = course
+        self.max_errors = 0
+
+    @property
+    def import_succeeded(self):
+        """
+        Helper wrapper that tells us the status of the import
+        """
+        return not self.validation_errors
+
+    def set_team_membership_from_csv(self, input_file):
+        """
+        Assigns team membership based on the content of an uploaded CSV file.
+        Returns true if there were no issues.
+        """
+        self.course_module = modulestore().get_course(self.course.id)
+        reader = csv.DictReader((line.decode('utf-8-sig').strip() for line in input_file.readlines()))
+        self.teamset_ids = reader.fieldnames[2:]
+        row_dictionaries = []
+        if self.validate_teamsets():
+            # process student rows:
+            for row in reader:
+                username = row['user']
+                if not username:
+                    continue
+                user = self.get_user(username)
+                if user is None:
+                    continue
+                if not self.validate_user_enrolled_in_course(user):
+                    row['user'] = None
+                    continue
+                row['user'] = user
+
+                if not self.validate_user_assignment_to_team_and_teamset(row):
+                    return False
+                row_dictionaries.append(row)
+
+            if not self.validation_errors:
+                for row in row_dictionaries:
+                    self.add_user_to_team(row)
+                return True
+            else:
+                return False
+        return False
+
+    def validate_teamsets(self):
+        """
+        Validates team set names. Returns true if there are no errors.
+        The following conditions result in errors:
+        Teamset does not exist
+        Teamset id is duplicated
+        Also populates the teamset_names_list.
+        header_row is the list of teamset_ids
+        """
+        teamset_ids = {ts.teamset_id for ts in self.course_module.teams_configuration.teamsets}
+        dupe_set = set()
+        for teamset_id in self.teamset_ids:
+            if teamset_id in dupe_set:
+                self.validation_errors.append("Teamset with id " + teamset_id + " is duplicated.")
+                return False
+            dupe_set.add(teamset_id)
+            if teamset_id not in teamset_ids:
+                self.validation_errors.append("Teamset with id " + teamset_id + " does not exist.")
+                return False
+            self.user_ids_by_teamset_id[teamset_id] = {m.user_id for m in CourseTeamMembership.objects.filter
+                                                       (team__course_id=self.course.id, team__topic_id=teamset_id)}
```

Comment:
Nit:  this would be cleaner as
```python
self.user_ids_by_teamset_id[teamset_id] = {
    membership.user_id for membership in 
    CourseTeamMembership.objects.filter(
        team__course_id=self.course.id, team__topic_id=teamset_id
    )
}
```